BUILD.md

```diff
@@ -0,0 +1,88 @@
+# Building DLAF
+
+## Build using Spack
+
+Example installation:
+
+`spack install dla-future ^[virtuals=blas,lapack] intel-oneapi-mkl`
+
+or you can also build C API compatible with ScaLAPACK with:
+
+`spack install dla-future +scalapack ^[virtuals=blas,lapack,scalapack] intel-oneapi-mkl`
+
+You can go even further with a more detailed spec like this one, which builds dla-future in debug mode, using the clang compiler, specifying that the pika on which it depends has to be built
+in debug mode too, and that we want to use MPICH as MPI implementation, without fortran support (because clang does not support it).
+
+`spack install dla-future %clang build_type=Debug ^pika build_type=Debug ^mpich ~fortran`
+
+### master branch
+
+If you want to test the newest available version,
+you can use the spack package `dla-future` provided in the git repository, that can be easily added to your own spack as follows:
```

Comment:
Should we note that `spack@master` can also be installed from the upstream spack package definition?

Reply:
Fair point, let's keep it like this.